src/utils/error-types.ts

```diff
@@ -0,0 +1,41 @@
+class ValidationError extends Error {
```

Comment:
Can move all exceptions to their separate files and move all those files to a separate `exceptions` module